### ISSUE TITLE
Multipe Network I/Fs and DCHP

### ISSUE THREAD

**Issue by joakim-tjernlund** [ipxe/ipxe#677]
When more that one I/F, ipxe probes them in sequence so one must wait for ipxe to get around to later I/Fs
Maybe ipxe can do what Linux ipconfig does, probe all I/F in parallel and use the first one that gets a link?

**Comment by mcb30** [comment]
Another issue is that iPXE runs with very limited memory available for I/O buffers, and opening all network cards at once is likely to lead to memory exhaustion and large numbers of dropped packets.

**Comment by joakim-tjernlund** [comment]
> Another issue is that iPXE runs with very limited memory available for I/O buffers, and opening all network cards at once is likely to lead to memory exhaustion and large numbers of dropped packets.

How so? ipxe would only poll all I/Fs for link up, when it finds one ipxe stops polling the other ones and continue with the with link up.

**Comment by stappersg** [comment]
@joakim-tjernlund what about proper spelling of DHCP?

![image](https://user-images.githubusercontent.com/35434094/183697273-2a7649ea-065e-4f8d-9207-c49037778b8d.png)

**Comment by mcb30** [comment]
> > Another issue is that iPXE runs with very limited memory available for I/O buffers, and opening all network cards at once is likely to lead to memory exhaustion and large numbers of dropped packets.
> 
> How so? ipxe would only poll all I/Fs for link up, when it finds one ipxe stops polling the other ones and continue with the with link up.

There is no concept within iPXE of opening an interface without providing it with receive data buffers, and no way of polling for link state without opening the interface.

It could potentially be done, but it would have to be a substantial benefit to justify the amount of work.

Marking as closed on the basis that this is unlikely to be carried out.
